### DISCUSSION TITLE
Slice to volume registration

### DISCUSSION THREAD

**Discussion by alexeytopolnitskiy** [dipy/dipy#3576] [1. Ideas & Feedback]
Hello!

I am working on the project with CT images and I have a relevant question.
I have a 3D model of CT images and 2D SEM image. And I need to find a 2D CT slice inside 3D CT model which as close as possible to original 2D SEM image, i.e. make slice to volume registration. 
As an output I need a registered 2D CT slice from 3D model or at least transformation parameters.

Is it possible to do with dipy?

Best, Alexey

**Comment by Garyfallidis** [comment]
Not yet, but we are working on it! Can you provide a few more details?

**Comment by alexeytopolnitskiy** [comment]
I am working with geological data. And I want to find a SEM image inside a 3D CT model. You can imagine a 3D model as just a numpy.ndarray. 

I created a simple algorithm that rotates 3D model in different directions and calculates the mutual information score between each CT slice and SEM image. And the best value gives the appropriate type of transformation (angles in three directions). But when I get these transformations I start to struggle with obtaining the best CT slice because simple opencv rotations are not able to perform right transformation, it just rotates in every direction and in the end the final slice is not correct.

So I am still in the process of finding the solution.
